radiation/python/radsw/vrtqdr_gt4py.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import xarray as xr
+import os
+import sys
+from gt4py.gtscript import stencil, computation, interval, PARALLEL, BACKWARD, mod
+
+sys.path.insert(0, "/Users/AndrewP/Documents/work/physics_standalone/radiation/python")
+
+from util import compare_data, create_storage_from_array, create_storage_zeros
+from config import *
+
+rebuild = False
+validate = True
+backend = "gtc:gt:cpu_ifirst"
+
+SERIALBOX_DIR = "/Users/AndrewP/Documents/code/serialbox2/install"
+sys.path.append(SERIALBOX_DIR + "/python")
+import serialbox as ser
+
+ddir = "/Users/AndrewP/Documents/work/physics_standalone/radiation/fortran/radsw/dump"
+
+serializer = ser.Serializer(ser.OpenModeKind.Read, ddir, "Serialized_rank1")
+savepoints = serializer.savepoint_list()
+
+invars = [
+    "zrefb",
+    "zrefd",
+    "ztrab",
+    "ztrad",
+    "zldbt",
+    "ztdbt",
+]
+
+outvars = ["zfu", "zfd"]
+
+locvars = ["zrupb", "zrupd", "zrdnd", "ztdn", "zden1"]
+
+indict = dict()
+for var in invars:
+    tmp = serializer.read(var, serializer.savepoint["swrad-vrtqdr-input-000000"])
+    indict[var] = np.tile(tmp[None, None, :], (npts, 1, 1))
+
+indict_gt4py = dict()
+for var in invars:
+    indict_gt4py[var] = create_storage_from_array(
+        indict[var], backend, shape_nlp1, DTYPE_FLT
+    )
+
+outdict_gt4py = dict()
+locdict_gt4py = dict()
+
+for var in outvars:
+    outdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+
+for var in locvars:
+    locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+
+
+@stencil(backend=backend, rebuild=rebuild)
+def vrtqdr(
+    zrefb: FIELD_FLT,
+    zrefd: FIELD_FLT,
+    ztrab: FIELD_FLT,
+    ztrad: FIELD_FLT,
+    zldbt: FIELD_FLT,
+    ztdbt: FIELD_FLT,
+    zfu: FIELD_FLT,
+    zfd: FIELD_FLT,
+    zrupb: FIELD_FLT,
+    zrupd: FIELD_FLT,
+    zrdnd: FIELD_FLT,
+    ztdn: FIELD_FLT,
+    zden1: FIELD_FLT,
+):
+
+    with computation(PARALLEL), interval(0, 1):
+        # Link lowest layer with surface.
+        zrupb = zrefb  # direct beam
+        zrupd = zrefd  # diffused
+
+    with computation(PARALLEL), interval(1, None):
+        # Pass from bottom to top.
+        zden1 = 1.0 / (1.0 - zrupd[0, 0, -1] * zrefd)
+        zrupb = (
+            zrefb
+            + (ztrad * ((ztrab - zldbt) * zrupd[0, 0, -1] + zldbt * zrupb[0, 0, -1]))
+            * zden1
+        )
+        zrupd = zrefd + ztrad * ztrad * zrupd[0, 0, -1] * zden1
```

Comment:
Can these two be combined into one computation?

Reply:
Not sure, will discuss